metricbeat/helper/prometheus/prometheus.go

```diff
@@ -63,19 +64,31 @@ func NewPrometheusClient(base mb.BaseMetricSet) (Prometheus, error) {
 	}
 
 	http.SetHeaderDefault("Accept", acceptHeader)
+	http.SetHeader("Accept-Encoding", "gzip")
```

Comment:
Could you use `SetHeaderDefault`? The reason is that this way, the header can be overridden by users from the config